### PR TITLE
[DataCollector] Fix freeze on some container

### PR DESCRIPTION
**Pull request by valouvaliavlo**
OctoBot WebInterface will freeze with a large data (when data is saved to db) so the container will restart If it's is set to restart on fail

### CODE REVIEW

**Comment by valouvaliavlo** [comment]
Freeze with digitalocean 5€ plan for example

File: Backtesting/collectors/exchanges/exchange_history_collector/history_collector.py [hunk 1]
```diff
@@ -147,17 +158,27 @@ async def get_ohlcv_history(self, exchange, symbol, time_frame):
...
+
+                    if self.end_timestamp is not None:
+                        while candles[-1][commons_enums.PriceIndexes.IND_PRICE_TIME.value] > self.end_timestamp:
+                            candles.pop(-1)
+                    self.exchange.uniformize_candles_if_necessary(candles)
+                    await self.save_ohlcv(exchange=exchange,
+                                  cryptocurrency=self.exchange_manager.exchange.get_pair_cryptocurrency(symbol),
+                                  symbol=symbol, time_frame=time_frame, candle=candles,
+                                  timestamp=[candle[0] + time_frame_sec for candle in candles], multiple=True)
+                    candles.clear()
```
Comment (GuillaumeDSM) [inline Backtesting/collectors/exchanges/exchange_history_collector/history_collector.py:172]: the idea here is to avoid writing too many data at once ?

Comment (valouvaliavlo) [reply Backtesting/collectors/exchanges/exchange_history_collector/history_collector.py:172]: yes, right

Comment (GuillaumeDSM) [reply Backtesting/collectors/exchanges/exchange_history_collector/history_collector.py:172]: and there is no issue in making multiple inserts like this ? I mean no double saved candles etc ?

Comment (valouvaliavlo) [reply Backtesting/collectors/exchanges/exchange_history_collector/history_collector.py:172]: It does exactly the same thing. 
Unit test is ok. 
Do you want a unit test with larger daterange on original version to compare with it ?

Comment (GuillaumeDSM) [reply Backtesting/collectors/exchanges/exchange_history_collector/history_collector.py:172]: Right, it tests are ok, it must be fine !

**Review by GuillaumeDSM — APPROVED** [review]
👍

**Comment by GuillaumeDSM** [comment]
Hi @valouvaliavlo, we merged the last PR with your new tests, when this one will be rebased on dev, we will be able to merge it !

**Comment by valouvaliavlo** [comment]
> Hi @valouvaliavlo, we merged the last PR with your new tests, when this one will be rebased on dev, we will be able to merge it !

done !

**Review by Herklos — APPROVED** [review]
:+1:

**Pull request merged**